sql: support synchronous_commit and enable_seqscan as dummy no-op

otan opened the pull request:
These vars are set by `osm2pgsql` and `ogr2ogr` respectively. These
default to the ON state, the OFF state affects performance but not
correctness.

Touches #51818.

Release note (sql change): Support the setting and getting of the
`synchronous_commit` and `enable_seqscan` variables, which do not affect
any performance characteristics. These are no-ops enabled to allow
certain tools to work.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/52168)
<!-- Reviewable:end -->

rytaft reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 8 of 8 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/52168#-:-MDp8aOS6wjGE_roGTOI:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @otan)

___
*[pkg/sql/unsupported_vars.go, line 96 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/52168#-MDp5MEb814NO9zXd9nr:-MDp5MEc39Z-X5JyiqT7:b-fnnghq) ([raw file](https://github.com/cockroachdb/cockroach/blob/51e9dc46c6b81a9c154cbab9b8586821263bbc86/pkg/sql/unsupported_vars.go#L96)):*
> ```Go
> 	"enable_mergejoin",
> 	"enable_nestloop",
> 	//"enable_seqscan",
> ```

[nit] add a space after //
___
*[pkg/sql/logictest/testdata/logic_test/set, line 360 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/52168#-MDp6jCJC9XJtMdZ42f1:-MDp6jCJC9XJtMdZ42f2:bwm0xmb) ([raw file](https://github.com/cockroachdb/cockroach/blob/51e9dc46c6b81a9c154cbab9b8586821263bbc86/pkg/sql/logictest/testdata/logic_test/set#L360)):*
> ```
> 
> query T noticetrace
> SET synchronous_commit = off; SET enable_seqscan = false
> ```

what about also adding a test for enabling these (setting to true)? Would be good to add a test to `SHOW` the current value as well.


<!-- Sent from Reviewable.io -->

rytaft reviewed (approved):
> Reviewed 3 of 3 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/52168)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @otan)

___
*[pkg/sql/logictest/testdata/logic_test/set, line 380 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/52168#-MDpMUCm7rULY-vC6_c3:-MDpMUCm7rULY-vC6_c4:birdf22) ([raw file](https://github.com/cockroachdb/cockroach/blob/fd359911666255234b25fa71278aa03b44106137/pkg/sql/logictest/testdata/logic_test/set#L380)):*
> ```
> 
> query T noticetrace
> SET synchronous_commit = off; SET enable_seqscan = true
> ```

[nit] maybe you meant synchronous_commit = on?


<!-- Sent from Reviewable.io -->

otan reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/52168)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @rytaft)

___
*[pkg/sql/logictest/testdata/logic_test/set, line 360 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/52168#-MDp6jCJC9XJtMdZ42f1:-MDpN0e_5vlPI7oZFS8l:bajeyq5) ([raw file](https://github.com/cockroachdb/cockroach/blob/51e9dc46c6b81a9c154cbab9b8586821263bbc86/pkg/sql/logictest/testdata/logic_test/set#L360)):*
<details><summary><i>Previously, rytaft (Rebecca Taft) wrote…</i></summary><blockquote>

what about also adding a test for enabling these (setting to true)? Would be good to add a test to `SHOW` the current value as well.
</blockquote></details>

caught a few things, oops. thanks for the suggestion
___
*[pkg/sql/logictest/testdata/logic_test/set, line 380 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/52168#-MDpMUCm7rULY-vC6_c3:-MDpN4XW6uS0Xi08xZRJ:b2nigkw) ([raw file](https://github.com/cockroachdb/cockroach/blob/fd359911666255234b25fa71278aa03b44106137/pkg/sql/logictest/testdata/logic_test/set#L380)):*
<details><summary><i>Previously, rytaft (Rebecca Taft) wrote…</i></summary><blockquote>

[nit] maybe you meant synchronous_commit = on?
</blockquote></details>

ye!


<!-- Sent from Reviewable.io -->

rytaft reviewed (approved):
> Reviewed 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/52168)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)



<!-- Sent from Reviewable.io -->

otan committed:
sql: support synchronous_commit and enable_seqscan as dummy no-op

These vars are set by `osm2pgsql` and `ogr2ogr` respectively. These
default to the ON state, the OFF state affects performance but not
correctness.

Release note (sql change): Support the setting and getting of the
`synchronous_commit` and `enable_seqscan` variables, which do not affect
any performance characteristics. These are no-ops enabled to allow
certain tools to work.

otan commented:
tftr!

bors r=rytaft
